source/install/trouble_mysql.rst

```diff
@@ -0,0 +1,268 @@
+issues.
+
+Before proceeding, confirm that your MySQL server is running. You can do
+this by issuing the following command: ::
+
+    mysqladmin -u root -p status
+
+When prompted, enter your password. If MySQL is running you should see output
+like the following: ::
+
+    Uptime: 877134  Threads: 1  Questions: 9902  Slow queries: 0  Opens: 522  
+    Flush tables: 1  Open tables: 371  Queries per second avg: 0.011
+
+If MySQL is not running, review the instructions for installation on
+your distribution.
+
+.. warning:: Some of the commands used in this section alter the database. **Use these commands only if your Mattermost installation has failed. Do not directly manipulate the MySQL database for a working Mattermost installation**.
+
+The mattermost Database
+~~~~~~~~~~~~~~~~~~~~~~~
+
+The database created during installation should be named ``mattermost``. If you
+fail to create this database or you misname it, when you attempt to
+start the Mattermost server you will see an error such as: ::
+
+    [2017/09/20 17:11:37 EDT] [INFO] Pinging SQL master database
+    [2017/09/20 17:11:37 EDT] [EROR] Failed to ping DB retrying in 10 seconds
+    err-Error 1044: Access denied for user 'mmuser'@'%' to database 'mattermost'
+
+Note that MySQL is specifically denying access to the ``mattermost``
+database. This may mean that you have failed to create a database named
+``mattermost`` or you may have incorrectly referenced this database from
+the ``/opt/mattermost/config/config.json`` file.
+
+**Checking that the Database Exists**
+
+To confirm that the ``mattermost`` database exists, open MySQL as root
+by executing ``mysql -u root -p``. When prompted, enter your
+password and then issue the following command: ::
+
+    show databases;
+
+This command displays all the databases. You should see something similar to the
+following: ::
+
+    +--------------------+
+    | Database           |
+    +--------------------+
+    | information_schema |
+    | mattermost         |
+    | mysql              |
+    | performance_schema |
+    | sys                |
+    +--------------------+
+    5 rows in set (0.03 sec)
+
+**No mattermost Database**
+
+If the ``mattermost`` database doesn't exist, create a database named
+``mattermost`` by opening MySQL as root and issuing the command: ::
+
+    create database mattermost;
+
+If you accidentally created a database with the wrong name, you can
+remove it by issuing the command: ::
+
+    drop database {misnamed};
+
+After creation of the database, attempt to restart the Mattermost server
+by navigating to the ``/opt/mattermost/bin`` directory and entering the
+command ``sudo -u mattermost ./platform``.
+
+**The mattermost Database Exists**
+
+If the ``mattermost`` database does exist, confirm that you have set up
+the database driver correctly in the
+``/opt/mattermost/config/config.json`` file. Open this file in a text
+editor, and review the value of ``"DataSource"``. It should be: ::
```

Comment:
We usually refer to these settings by their full name so this would be `DatabaseSettings.DataSource` and below we'd have `DatabaseSettings.DriverName`